src/libs/Localize/index.ts

```diff
@@ -104,14 +108,40 @@ function getTranslatedPhrase<TKey extends TranslationPaths>(
     const valueFromCache = cacheForLocale?.get(phraseKey);
 
     // If the phrase is already translated, return the translated value
-    if (valueFromCache) {
+    if (valueFromCache && typeof valueFromCache === 'string') {
         return valueFromCache;
     }
 
     const translatedPhrase = translations?.[language]?.[phraseKey] as Phrase<TKey>;
 
     if (translatedPhrase) {
         if (typeof translatedPhrase === 'function') {
+            const result = translatedPhrase(...phraseParameters);
+            const countParam = phraseParameters.find((param: unknown) => !!(param as Record<string, number>).count) as Record<string, number>;
+            const count = countParam ? countParam.count : undefined;
+
+            if (typeof result === 'object' && count !== undefined) {
+                const pluralRules = new Intl.PluralRules(language);
+                const pluralForm = pluralRules.select(count);
+
+                if (pluralForm in result) {
+                    const translatedPluralForm = result[pluralForm as keyof typeof result];
+                    if (typeof translatedPluralForm === 'string') {
+                        return translatedPluralForm;
+                    }
+                }
+
+                Log.warn(`Pluralization form '${pluralForm}' not found for phrase '${phraseKey}'. Falling back to 'other' form.`);
```

Comment:
I don't think this is correct, is it?
If the pluralForm key exists in result but it is a function (not a string) then it will warn, but that's not what we want.